app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+
+                // if the allergen commonName or inciName appears in the ingredient inciName or
+                // description respectively
+                if(ingr.getInciName().contains(allergenInciName) ||
+                        ingr.getDescription().toUpperCase().contains(allergen.getCommonName())) {
+
+                    // add the allergen to foundAllergens
+                    foundAllergens.add(allergen);
+
+                    // if the allergen is found go to the next one
+                    break;
+                }
+            }
+        }
+        return foundAllergens;
+    }
+
+    /**
+     * Check if an ingredient contains a selected allergen common name or inci name (stored in SharedPreferences sp)
+     * @param ingr, ingredient to check for allergens
+     * @return the allergen found in the ingredient, empty ArrayList if no allergen is found
+     * @author Nicolò Cervo (g3)
+     */
+    public ArrayList<Allergen> checkForSelectedAllergens(Ingredient ingr) {
+
+        Map<String, ?> selectedAllergens = sp.getAll();
+        Allergen allergen = new Allergen();
+        ArrayList<Allergen> foundSelectedAllergens = new ArrayList<>();
+
+        // For every selected allergen
+        for(Map.Entry<String, ?> entry : selectedAllergens.entrySet()) {
+            // Check if the ingredient has one of it's inci names or it's common name(in the description)
+            for(String inciName : entry.getValue().toString().split(", ")) {
+                if(ingr.getInciName().contains(inciName) ||
+                        ingr.getDescription().toUpperCase().contains(entry.getKey())) {
+
+                    allergen.setInciNames(entry.getValue().toString().split(", "));
+                    allergen.setCommonName(entry.getKey());
+                    foundSelectedAllergens.add(allergen);
+                    break;
+                }
+            }
+        }
+        return foundSelectedAllergens;
+    }
+
+    /**
+     * Checks an ArrayList of ingredients for allergens.
+     * @param ingrList, List of ingredients to check.
+     * @return foundAllergens, ArrayList of allergens found in ingrList.
+     * @author Nicolò Cervo (g3)
+     */
+    public ArrayList<Allergen> checkListForAllergens(List<Ingredient> ingrList) {
+        ArrayList<Allergen> foundAllergens = new ArrayList<>();
+
+        for(Ingredient ingr : ingrList) {
+            foundAllergens.addAll(checkForAllergens(ingr));
+        }
+        return foundAllergens;
+    }
+
+    /**
+     * Checks an ArrayList of ingredients for selected allergens.
+     * @param ingrList, List of ingredients to check.
+     * @return foundAllergens, ArrayList of selected allergens found in ingrList.
+     * @author Nicolò Cervo (g3)
+     */
+    //Todo: check if this method is correct, used int GalleryManager onBindViewHolder
+    public ArrayList<Allergen> checkListForSelectedAllergens(List<Ingredient> ingrList) {
```

Comment:
Similarly to the previous comment, I would suggest you to overload the method checkForSelectedAllergens()